db/db.go

```diff
@@ -314,19 +318,24 @@ func InitalizeMetadataDbConnection(l logger) (MetadataStorage, error) {
 	config.MaxConns = 5
 
 	if configuration.METADATA_DB_TLS_ENABLED {
-		cert, err := tls.LoadX509KeyPair(configuration.METADATA_DB_TLS_CRT, configuration.METADATA_DB_TLS_KEY)
-		if err != nil {
-			return MetadataStorage{}, err
-		}
-
 		CACert, err := os.ReadFile(configuration.METADATA_DB_TLS_CA)
 		if err != nil {
 			return MetadataStorage{}, err
 		}
 
 		CACertPool := x509.NewCertPool()
 		CACertPool.AppendCertsFromPEM(CACert)
-		config.ConnConfig.TLSConfig = &tls.Config{Certificates: []tls.Certificate{cert}, RootCAs: CACertPool, InsecureSkipVerify: true}
+
+		if configuration.METADATA_DB_TLS_MUTUAL {
+			cert, err := tls.LoadX509KeyPair(configuration.METADATA_DB_TLS_CRT, configuration.METADATA_DB_TLS_KEY)
```

Comment:
Overall looks ok, just one question, please.
In the case of METADATA_DB_TLS_MUTUAL=false you are not using the configuration.METADATA_DB_TLS_CRT at all. This is actually changing the current behavior, could you please explain why?

Reply:
@idanasulinmemphis this is so that memphis can be used with an external, existing Postgres server that does not support Mutual TLS (or where this is not desired). 

The goal is to not provide a client cert and key, but instead use password auth for the DB, while still using an encrypted connection and verifying the server's certs.

A specific use case is Amazon Aurora RDS for Postgres with password auth